pysonde/readers/readers.py

```diff
@@ -150,35 +150,44 @@ def read(self, cor_file, bufr_file=None, round_like_bufr=False):
         pd_snd = pd.read_csv(cor_file, delimiter="\t")
 
         def _get_date_information_from_filename(cor_file):
+            """
+            Extracts the launch date from a filename, adjusting for forecast hour shifts.
+
+            Args:
+                cor_file (str): Filename (e.g., 'SA2024090618_1.cor').
+                filename_fmt (dict): Dictionary with filename formatting rules.
+                pd_snd (xarray.Dataset or similar): Dataset containing time information.
```

Comment:
These are currently not arguments, but they actually should :) Could you add them as function arguments and adjust the function calls accordingly?